java/ruleXtract/src/main/java/uk/ac/cam/eng/extraction/hadoop/extraction/ExtractorJob.java

```diff
@@ -150,86 +161,44 @@ protected void map(MapWritable key, TextArrayWritable value,
 						ruleInfo.putProvenanceCount(prov2Id.get(prov), ONE);
 					}
 				}
-				ruleInfo.putAlignmentCount(aw, 1);
-				context.write(rw, ruleInfo);
+				ruleInfo.putAlignmentCount(ra.getSecond(), 1);
+				context.write(ra.getFirst(), ruleInfo);
 			}
 		}
 	}
 
-	private static class ExtractorReducer
-			extends
-			Reducer<RuleWritable, RuleInfoWritable, RuleWritable, RuleInfoWritable> {
+	private static class ExtractorReducer extends
+			Reducer<Rule, ExtractedData, Rule, ExtractedData> {
 
-		private RuleInfoWritable compressed = new RuleInfoWritable();
+		private ExtractedData compressed = new ExtractedData();
 
 		@Override
-		protected void reduce(
-				RuleWritable key,
-				Iterable<RuleInfoWritable> values, Context context)
-				throws IOException, InterruptedException {
+		protected void reduce(Rule key, Iterable<ExtractedData> values,
+				Context context) throws IOException, InterruptedException {
 			compressed.clear();
-			for (RuleInfoWritable value : values) {
+			for (ExtractedData value : values) {
 				compressed.increment(value);
 			}
 			context.write(key, compressed);
 		}
 	}
 
-	/**
-	 * Defines command line args.
-	 */
-	@Parameters(separators = "=")
-	public static class ExtractorJobParameters {
-		@Parameter(names = { "--input", "-i" }, description = "Input training data on HDFS", required = true)
-		public String input;
-
-		@Parameter(names = { "--output", "-o" }, description = "Output rules on HDFS", required = true)
-		public String output;
-
-		@Parameter(names = { "--remove_monotonic_repeats"}, description = "Gives an "
-				+ "occurrence count of 1 to monotonic hiero rules (e.g. "
-				+ "phrase-pair <a b c, d e f> with alignment 0-0 1-1 2-2 "
-				+ "generates hiero rule <a X, d X> twice but the count is "
-				+ "still one)")
-		public String remove_monotonic_repeats = "true";
-
-		@Parameter(names = { "--max_source_phrase"}, description = "Maximum source phrase length in a phrase-based rule")
-		public String max_source_phrase = "9";
-		
-		@Parameter(names = { "--max_source_elements"}, description = "Maximum number of source elements (terminals and nonterminals) in a hiero rule")
-		public String max_source_elements = "5";
-
-		@Parameter(names = { "--max_terminal_length" }, description = "Maximum number of consecutive source terminals in a hiero rule")
-		public String max_terminal_length = "5";
-
-		@Parameter(names = { "--max_nonterminal_span" }, description = "Maximum number of source terminals covered by a right-hand-side source nonterminal in a hiero rule")
-		public String max_nonterminal_span = "10";
-
-		@Parameter(names = { "--provenance" }, description = "Comma-separated provenances")
-		public String provenance;
-	}
-
 	public int run(String[] args) throws FileNotFoundException, IOException,
```

Comment:
This a comment attached to existing code. I haven't edited it. I agree that that it would be good to have constants but this is out of scope of the changes.